src/engine/Settings.cs

```diff
@@ -1,15 +1,26 @@
-﻿using Godot;
```

Comment:
I don't get what the changes in this file are about? What was wrong with the previous approach?

Reply:
Originally I was just going to do some simple changes for the json saving but I figured the more proper way to do it would be have it detect any changes to the running settings vs ones saved to the configuration file. Maybe a bit overkill but you can do things like toggle a setting on and then off now and it will disable the buttons since it matches the old settings.

Since the settings class had a private constructor and I didn't really want to mess with trying to duplicate it(so I could save the state for comparison) and I didn't really want to manually copy each variable and compare it. I opted to just split the data into a struct that could be accessed. That way I can just copy it and compare it to any live settings changes. Essentially when you mess with the options menu it is changing the Settings class singleton on the fly and applying any changes. If you reset then it copies back the old state, or if you hit save then it copies the new state to the file and replaces the old one.

I was trying to avoid changing the interface too much cause I figured that would require changes in a lot more places, that's why the properties are the same.

Reply:
I would suggest that you undo the change to have the setting values in the struct.
Instead add a new method like (or implement the clonable interface):
```
public Settings CopyCurrentValues()
{
    // Create new settings object here and use reflection to grab the values of all fields, that way this is automatically up to date when new settings are added
}
```
That would allow the options menu to take a snapshot of the values for comparison purposes.

Reply:
Here's an existing method that the json saving and loading system uses: https://github.com/Revolutionary-Games/Thrive/blob/0e79166eedcc408980e2215b63cba2c37acb926b/src/saving/serializers/ThriveJsonConverter.cs#L241
that can be used to get the properties for looping through and copying.

Reply:
It's basically a stylistic choice so I can change it where you can copy/clone Settings instead, that's fine.